internal/webserver/webserver.go

```diff
@@ -306,8 +306,15 @@ func (n *kubeFilter) Start(ctx context.Context) error {
 			tlsConfig := &tls.Config{
 				MinVersion: tls.VersionTLS12,
 				ClientCAs:  n.serverOptions.GetCertificateAuthorityPool(),
-				ClientAuth: tls.VerifyClientCertIfGiven,
+				ClientAuth: tls.NoClientCert,
 			}
+
+			for _, authType := range n.authTypes {
+				if authType == req.TLSCertificate {
+					tlsConfig.ClientAuth = tls.VerifyClientCertIfGiven
```

Comment:
Once found, we should `break` the iteration.

Reply:
added break statement. thank you!